magicgui/widgets/_bases/widget.py

```diff
@@ -90,7 +92,22 @@ def __init__(
         _protocols.assert_protocol(widget_type, prot)
         self.__magicgui_app__ = use_app()
         assert self.__magicgui_app__.native
-        self._widget = widget_type(**backend_kwargs)
+        if isinstance(parent, Widget):
+            parent = parent.native
+        try:
+            self._widget = widget_type(parent=parent, **backend_kwargs)
```

Comment:
👀 @hanjinliu  ... this is the specific line that would break magic-class.  There's a fallback to the old behavior with a warning below